private/topology/linktype.go

```diff
@@ -59,17 +61,21 @@ func (l LinkType) String() string {
 
 // LinkTypeFromString returns the numerical link type associated with a string description. If the
 // string is not recognized, an Unset link type is returned. The matching is case-insensitive.
+// @ requires low(s)
 // @ decreases
 func LinkTypeFromString(s string) (res LinkType) {
 	var l /*@@@*/ LinkType
 	tmp := []byte(s)
+	// TODO: Once Gobra issue #831 is resolved, remove this assumption.
+	//@ assume forall i int :: { &tmp[i] } 0 <= i && i < len(tmp) && low(i) ==> low(tmp[i])
```

Comment:
to avoid the proliferation of these kinds of assumptions throughout the codebase (regardless of how benign they are), I would rather introduce a lemma here that gives me this postcondition.